src/worker/exporters/node_exporter.py

```diff
@@ -247,7 +248,23 @@ def init_config(job_id, port=None):
         'fieldFiles': {},
         'counter': {},
     }
-
+    # for xid and link flaps
+    config['command'] = {}
+    
+    cmd ="awk -F= '/^NAME/{print $2}' /etc/os-release"
+    result = shell_cmd(cmd, 5)
+    if "Ubuntu" in result:
+        config['command']['link_flap'] = "grep 'Lost carrier' /var/log/syslog"
+        config['command']['xid_error'] = "grep 'NVRM: Xid' /var/log/syslog"
+        init_nvidia_config()
+        init_ib_config()
+    elif "AlmaLinux" in result:
+        config['command']['link_flap'] = "sudo grep 'Lost carrier' /var/log/messages"
+        config['command']['xid_error'] = "sudo grep 'NVRM: Xid' /var/log/messages"
+        init_nvidia_config()
+        init_ib_config()
+    else:
+        logging.info('OS not supported attempting to continue...')
```

Comment:
Is there a point in attempting to continue if collect() is going to try and access a nonexistent dict entry?